service/api.go

```diff
@@ -106,14 +107,27 @@ func (s *gnomobileService) SelectAccount(ctx context.Context, req *connect.Reque
 	}
 
 	s.lock.Lock()
-	s.activeAccount = key
+	account, ok := s.userAccounts[req.Msg.NameOrBech32]
+	if !ok {
+		account = &userAccount{}
+		s.userAccounts[req.Msg.NameOrBech32] = account
+	}
+	account.keyInfo = key
+	s.activeAccount = account
 	s.lock.Unlock()
 
 	s.getSigner().Account = req.Msg.NameOrBech32
+	s.getSigner().Password = account.password
 	return connect.NewResponse(&rpc.SelectAccountResponse{Key: info}), nil
 }
 
 func (s *gnomobileService) SetPassword(ctx context.Context, req *connect.Request[rpc.SetPasswordRequest]) (*connect.Response[rpc.SetPasswordResponse], error) {
+	s.lock.Lock()
+	if s.activeAccount != nil {
+		s.activeAccount.password = req.Msg.Password
+	}
+	s.lock.Unlock()
+
 	s.getSigner().Password = req.Msg.Password
```

Comment:
Is this useful if we allow setting a password without an active account?

Reply:
As agreed, we don't allow this.